src/libra_py/workflows/nbra/rpi.py

```diff
@@ -0,0 +1,433 @@
+# *********************************************************************************
+# * Copyright (C) 2025 Daeho Han and Alexey V. Akimov
+# *
+# * This file is distributed under the terms of the GNU General Public License
+# * as published by the Free Software Foundation, either version 3 of
+# * the License, or (at your option) any later version.
+# * See the file LICENSE in the root directory of this distribution
+# * or <http://www.gnu.org/licenses/>.
+# *
+# *********************************************************************************/
+
+"""
+.. module:: rpi
+   :platform: Unix, Windows
+   :synopsis: this module implements restricted path integral (RPI) calculations
+
+.. moduleauthor:: Daeho Han, Alexey V. Akimov
+
+"""
+
+import os
+import sys
+import numpy as np
+import scipy.sparse as sp
+import h5py
+import time
+import multiprocessing as mp
+if sys.platform == "cygwin":
+    from cyglibra_core import *
+elif sys.platform == "linux" or sys.platform == "linux2":
+    from liblibra_core import *
+import libra_py.units as units
+from libra_py import data_conv
+import libra_py.data_read as data_read
+import util.libutil as comn
+import libra_py.dynamics.tsh.compute as tsh_dynamics
+
+def run_patch_dyn_serial(rpi_params, istep, fstep, istate):
+    """
+    This function runs a single patch dynamics calculation for the time segment from istep to fstep and a given initial state.
+    For conducting a coherent Ehrenfest propagation, the `libra_py.dynamics.tsh.compute` module is used.
+
+    Args:
+
+        rpi_params ( dictionary ): parameters controlling the execution of the RPI dynamics
+            Can contain:
+
+            * **rpi_params["path_to_save_Hvibs"]** ( string ): The path of the vibronic Hamiltonian files.
+            
+            * **rpi_params["basis_type"]** ( string ): The electronic basis type (such as the Slater determinant or CI adpatation, i.e., 'sd' or 'ci').
+
+            * **rpi_params["dt"]** ( double ): the time step in the atomic unit.
```

Comment:
the default values should be defined for all the non-critical parameters and also listed in the documentation. (for critical ones - the user should defined them or the `check_input` function would stop the execution with an error)